### DISCUSSION TITLE
Assign specific configuration to specific source code file

### DISCUSSION THREAD

**Discussion by jy-roda** [microsoft/vscode-cpptools#10155] [Q&A]
Hello !
In my project, I have several configurations in my c_cpp_properties.json file (working on embedded stuff, software is separated in two memory zones for both code and data, but is still compiled in one final binary, and zones are interacting via a proxy).

Each C file is linked to one configuration (only one). Is there any way to auto switch configuration based on which file is focused on the editor, or any way for a file to use a specific configuration ?

there is another question related to my issue (I could assign folders also, there are fortunately no mixed folders, but it would be very tedious as there are a lot of them) : https://github.com/microsoft/vscode-cpptools/discussions/8226

I would need something like that.

Thank you :)

**Comment by sean-mcmanus** [comment]
You could create a compile_commands.json file and set it with the compileCommands property.
